plasmapy/physics/parameters.py

```diff
@@ -399,18 +399,112 @@ def thermal_speed(T, particle="e", method="most_probable"):
 
     # different methods, as per https://en.wikipedia.org/wiki/Thermal_velocity
     if method == "most_probable":
-        V = (np.sqrt(2*k_B*T/m)).to(units.m/units.s)
+        V = (np.sqrt(2 * k_B * T / m)).to(units.m / units.s)
     elif method == "rms":
-        V = (np.sqrt(3*k_B*T/m)).to(units.m/units.s)
+        V = (np.sqrt(3 * k_B * T / m)).to(units.m / units.s)
     elif method == "mean_magnitude":
-        V = (np.sqrt(8*k_B*T/(m/np.pi))).to(units.m/units.s)
+        V = (np.sqrt(8 * k_B * T / (m / np.pi))).to(units.m / units.s)
     else:
         raise(ValueError("Method {} not supported in thermal_speed"
                          .format(method)))
 
     return V
 
 
+@check_relativistic
+@check_quantity({
+    'T': {'units': units.K, 'can_be_negative': False}
+})
+def kappa_thermal_speed(T, kappa, particle="e", method="most_probable"):
+    r"""Returns the most probable speed for a particle within a Kappa
+    distribution.
+
+    Parameters
+    ----------
+    T : Quantity
+        The particle temperature in either kelvin or energy per particle
+
+    kappa: Quantity
+        The kappa parameter is a dimensionless number which sets the slope
+        of the energy spectrum of suprathermal particles forming the tail
+        of the Kappa velocity distribution function. Kappa must be greater
+        than 3/2.
+
+    particle : string, optional
+        Representation of the particle species (e.g., 'p' for protons, 'D+'
+        for deuterium, or 'He-4 +1' for singly ionized helium-4),
+        which defaults to electrons.  If no charge state information is
+        provided, then the particles are assumed to be singly charged.
+
+    method : string, optional
+        Method to be used for calculating the thermal speed. Options are
+        'most_probable' (default), 'rms', and 'mean_magnitude'. 
+
+    Returns
+    -------
+    V : Quantity
+        particle thermal speed
+
+    Raises
+    ------
+    TypeError
+        The particle temperature is not a Quantity
+
+    UnitConversionError
+        If the particle temperature is not in units of temperature or
+        energy per particle
+
+    ValueError
+        The particle temperature is invalid or particle cannot be used to
+        identify an isotope or particle
+
+    UserWarning
+        If the particle thermal speed exceeds 10% of the speed of light, or
+        if units are not provided and SI units are assumed.
+
+    Notes
+    -----
+    The particle thermal speed is given by:
+
+    .. math::
+        V_{th,i} = \sqrt{(2 \kappa - 3)\frac{2 k_B T_i}{\kappa m_i}}
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> kappa_thermal_speed(5*u.eV, 4, 'p')
+    <Quantity 24467.878463594967 m / s>
+    """
+    # Checking thermal units
+    T = T.to(units.K, equivalencies=units.temperature_energy())
+    # must have kappa > 3/2 for distribution function to be valid
```

Comment:
Lately I've been trying to follow the advice laid out in the paper on [_Best Practices for Scientific Computing_  by Wilson et al. (2014)](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745).  One of their suggestions is to: "Refactor code in preference to explaining how it works."  A possibility here would be to include the information in this comment in the ValueError message, and then delete the comment.  That way the code would be more self-documenting.

Reply:
I can see that being cleaner here, though I don't know if I would agree with that sentiment in general.